### PR TITLE
Enable findbugs annotations

### PR DESCRIPTION
**Pull request by oleg-nenashev**
We want to have automatic FindBugs checks in the plugin pull request builders

### CODE REVIEW

File: src/main/java/hudson/matrix/MatrixProject.java [hunk 1]
```diff
@@ -727,7 +732,14 @@ public MatrixConfiguration getItem(Combination c) {
         return configurations.get(c);
     }
 
-    public File getRootDirFor(MatrixConfiguration child) {
+    /**
+     * Gets a root directory for the combination.
+     * Creates the whole directory hierarchy on-demand.
+     * @param combination
```
Comment (olivergondza) [inline src/main/java/hudson/matrix/MatrixProject.java]: typo: configuration. As long I am concerned we can omit this entirely, it does not add much value.

**Comment by olivergondza** [comment]
:+1:

File: src/main/java/hudson/matrix/MatrixProject.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/hudson/matrix/MatrixProject.java]: I was quite confused by the implementation. Getter method with the automatic creation of directories seems to be quite crazy for me. It's being misused even within the plugin

**Comment by oleg-nenashev** [comment]
@olivergondza @reviewbybees
I'd like to fix more issues. Would you prefer to see them in a separate pull-request?

File: src/main/java/hudson/matrix/MatrixProject.java (hunk 1 above)
Comment (olivergondza) [inline src/main/java/hudson/matrix/MatrixProject.java]: Sorry, I meant this part specifically: `@param combination`.

Comment (oleg-nenashev) [inline src/main/java/hudson/matrix/MatrixProject.java]: Anyway, I've fixed Javadocs

**Comment by olivergondza** [comment]
+1 on adding findbugs into build pipeline and addressing issues later.

File: src/main/java/hudson/matrix/MatrixBuild.java [hunk 2]
```diff
@@ -147,10 +148,21 @@ public String getShortUrl() {
...
-        public String getTooltip() {
+        /**
+         * Gets a tooltip from the item.
+         * @return Tooltip or null if it cannot be retrieved.
+         */
+        public @CheckForNull String getTooltip() {
             MatrixRun r = getRun();
-            if(r!=null) return r.getIconColor().getDescription();
-            Queue.Item item = Jenkins.getInstance().getQueue().getItem(getParent().getItem(combination));
+            if(r!=null) {
```
Comment (KostyaSha) [inline src/main/java/hudson/matrix/MatrixBuild.java]: space after if, if you already changing lines

File: src/main/java/hudson/tasks/test/TestResultAggregator.java [hunk 3]
```diff
@@ -53,8 +53,13 @@ public boolean startBuild() throws InterruptedException, IOException {
 
     @Override
     public boolean endRun(MatrixRun run) throws InterruptedException, IOException {
-        AbstractTestResultAction atr = run.getAction(AbstractTestResultAction.class);
-        if(atr!=null)   result.add(atr);
+        final AbstractTestResultAction atr = run.getAction(AbstractTestResultAction.class);
+        if (result == null) {
+            throw new IOException("The buid has not been started yet.");
+        }
+        if(atr != null) {
```
Comment (KostyaSha) [inline src/main/java/hudson/tasks/test/TestResultAggregator.java]: space please

**Comment by KostyaSha** [comment]
:+1:

**Comment by oleg-nenashev** [comment]
@reviewbybees done
Squashing commits

**Pull request merged**